src/main/pages/che-7/installation-guide/assembly_airgap-installation.adoc

```diff
@@ -0,0 +1,161 @@
+
+## Introduction
+By default, Eclipse Che's workspaces reference various external resources, whether they are docker images available on public registries, or sample projects hosted on GitHub.
+
+In some situations, it may be desirable to deploy Che in an environment where these external resources are not available (e.g. on a cluster that is not exposed to the public internet). Doing this requires a few customizations:
+
+* Certain images need to be built containing external resources (workspace plugins, projects)
+* All images to be used for running Che need to be pulled to a registry that is visible
+* Che needs to be configured to use these images
+
+The following sections describe the steps necessary to deploy Che in an airgapped Kubernetes or OpenShift cluster.
+
+[#offline-registries]
+## Building offline versions of the plugin and devfile registry
+[NOTE]
+====
+This section describes how to build _offline_ registry images. For additional information, see the main link:{site-baseurl}che-7/building-and-running-a-custom-registry-image/[building and running a custom registry image] doc.
+====
+
+The main purpose of building offline regsitry images is to ensure that workspaces can be started without relying on resources from the outside internet. For the plugin registry, this means building a registry that serves all plugin or extension artifacts in additiona to plugin metadata. Similarly, an offline devfile registry serves all sample projects referenced in devfiles as `zip` files.
+
+The plugin registry repositories contain scripts to make it easier to build custom images.
+
+
+### Offline plugin registry
+[source,bash]
+----
+git clone https://github.com/eclipse/che-plugin-registry.git <1>
+cd che-plugin-registry
+git checkout 7.3.0 // <2>
+# Build the image
+./build.sh --organization <my-org> \
+           --registry <my-registry> \
+           --tag <my-tag> \
+           --offline // <3>
+----
+<1> Clone the plugin registry repository
+<2> Check out version of registry to be deployed
+<3> Full options for the `build.sh` script can be found by passing parameter `--help`
+
+This build of the registry will build a docker image that contains all `.vsix` and `.theia` extensions that can be used in workspaces; since this results in a quite large image, a version of the registry that contains only the _latest_ version of each plugin can be build using the argument `--latest-only`.
+
+
+### Offline devfile registry
+[source,bash]
+----
+git clone https://github.com/eclipse/che-devfile-registry.git <1>
+cd che-devfile-registry
+git checkout 7.3.0 // <2>
+# Build the image
+./build.sh --organization <my-org> \
+           --registry <my-registry> \
+           --tag <my-tag> \
+           --offline // <3>
+----
+<1> Clone the devfile registry repository
+<2> Check out version of registry to be deployed
+<3> Full options for the `build.sh` script can be found by passing parameter `--help`
+
+
+## Pulling images used by Che to a private registry
```

Comment:
Pulling or pushing to a repo ?

Reply:
I suppose either works -- I chose 'pull' because you're pulling public images into a private repo.

Reply:
Sure. I understand your point. Thanks for the explanation.